suite/tests/CMakeLists.txt

```diff
@@ -4176,140 +4362,9 @@ endif ()
-  string(REGEX REPLACE "^SHORT::" "" run "${run}")
-  if (is_short OR TEST_LONG)
-    set(enabled ON)
-  endif (is_short OR TEST_LONG)
-
-  string(REGEX MATCHALL "^DEBUG::" is_debug "${run}")
-  string(REGEX REPLACE "^DEBUG::" "" run "${run}")
-  if (is_debug AND NOT DEBUG)
-    set(enabled OFF)
-  endif (is_debug AND NOT DEBUG)
-
-  string(REGEX MATCHALL "^X86::" is_x86 "${run}")
-  string(REGEX REPLACE "^X86::" "" run "${run}")
-  if (is_x86 AND NOT X86)
-    set(enabled OFF)
-  endif ()
-
-  string(REGEX MATCHALL "^X64::" is_x64 "${run}")
-  string(REGEX REPLACE "^X64::" "" run "${run}")
-  if (is_x64 AND NOT X64)
-    set(enabled OFF)
-  endif ()
-
-  string(REGEX MATCHALL "^WIN::" is_win "${run}")
-  string(REGEX REPLACE "^WIN::" "" run "${run}")
-  if (is_win AND UNIX)
-    set(enabled OFF)
-  endif (is_win AND UNIX)
-
-  string(REGEX MATCHALL "^LIN::" is_lin "${run}")
-  string(REGEX REPLACE "^LIN::" "" run "${run}")
-  if (is_lin AND WIN32)
-    set(enabled OFF)
-  endif (is_lin AND WIN32)
-
-  if (enabled)
-
-    string(REGEX MATCHALL "^ONLY::.*::" regex "${run}")
-    string(REGEX REPLACE "^ONLY::.*::" "" run "${run}")
-    string(REGEX REPLACE "^ONLY::(.*)::" "\\1" regex "${regex}")
-
-    if (NOT is_short)
-      # FIXME i#1807: fix the failing long suite tests so we can enable them once again
-      # in our nightly regression runs.  For now we disable the extra tests (but
-      # we leave the extra builds selected in runsuite.cmake) by throttling
-      # all long tests to only run ^common.
-      set(regex "^common")
-    endif ()
-
-    ops2testprefix(prefix "${run}")
-    list(FIND prior_prefixes "${prefix}" index)
-    if (${index} EQUAL -1)
-      list(APPEND prior_prefixes "${prefix}")
-    endif ()
-    foreach (test ${testlist_normal})
-      is_test_enabled_by_name(name_ok ${test} "${regex}")
-      if (name_ok)
-        run2testname(test_name "${run}" ${test})
-        torun_normal("${test_name}" ${test} "${run}")
-      endif ()
-    endforeach (test)
-    foreach (test ${testlist_ci})
-      is_test_enabled_by_name(name_ok ${test} "${regex}")
-      if (name_ok)
-        run2testname(test_name "${run}" ${test})
-        torun_ci("${test_name}" ${test} "${run}")
-      endif ()
-    endforeach (test)
-    foreach (test ${testlist_api})
-      is_test_enabled_by_name(name_ok ${test} "${regex}")
-      if (name_ok)
-        run2testname(test_name "${run}" ${test})
-        torun_api("${test_name}" ${test} "${run}")
-      endif ()
-    endforeach (test)
-    foreach (test ${testlist_runall})
-      is_test_enabled_by_name(name_ok ${test} "${regex}")
-      if (name_ok)
-        run2testname(test_name "${run}" ${test})
-        torun_runall("${test_name}" ${test} "${run}")
-      endif ()
-    endforeach (test)
-
-  endif (enabled)
-endforeach(run)
+# Remember that we do not reach this point for DR_HOST_NOT_TARGET, so do not
+# place anything here that should be shared across all configurations.
+set_up_test_lists()
```

Comment:
Is there a way to assert this in CMAKE? Maybe test and raise FATAL_ERROR.

Reply:
Not easily.  We could set a cmake var here, and check whether it's set in some of the tobuild() and other macros: but that wouldn't stop a test being added between the `#` line and the set, nor would it stop a torunonly() or other variant, nor some other config tweak or test property.  I think it's ok: adding new code here is going to be pretty rare and the only thing here is this comment and the Mac test list so it should be low risk of sthg bad happening.